frontend/dockerfile/dockerfile2llb/convert_runnetwork.go

```diff
@@ -0,0 +1,27 @@
+// +build dfrunnetwork
+
+package dockerfile2llb
+
+import (
+	"github.com/pkg/errors"
+
+	"github.com/moby/buildkit/frontend/dockerfile/instructions"
+	"github.com/moby/buildkit/solver/pb"
+)
+
+func dispatchRunNetwork(d *dispatchState, c *instructions.RunCommand) error {
+	network := instructions.GetNetwork(c)
+
+	switch network {
+	case instructions.NetworkDefault:
+		d.state = d.state.Network(pb.NetMode_UNSET)
```

Comment:
This is slightly wrong. These should not modify the state but set runoptions for current command, so that they are set only for the current process and not for the processes after it. Currently, this doesn't appear for network because the next command resets the network back to `NetMode_UNSET` but logically this isn't correct.

I noticed the same issue in the `--security` PR and there it actually causes wrong behavior where a command coming after `--security=insecure` also has that profile. Also the securitymode shouldn't be an array but only single value. @smira Can you take a look.

To avoid these issues lets add a testcases for network and security tests that check the profile of the next command as well.

Reply:
@smira - I can make the list -> string change if you like (and bring these two features into line) as I've already made a few minor changes to the `--security` code which might lead to tedious merge issues and the change to a single value is pretty trivial.

Reply:
Just to be clear, in addition to list->string change both security and network implementation should use https://godoc.org/github.com/moby/buildkit/client/llb#Network style function here returning `RunOption` and not modify the state with https://godoc.org/github.com/moby/buildkit/client/llb#State.Network

Reply:
I can add a test for `RUN security=` as a follow up PR, this is pretty trivial (to make sure it gets reset for the next `RUN` instruction)